diff --git a/SEISMIC.R b/SEISMIC.R
index 4ccbd11..06f47cb 100755
--- a/SEISMIC.R
+++ b/SEISMIC.R
@@ -1,5 +1,19 @@
 #!/usr/bin/env Rscript
 
+###############################################
+#                                             #
+#            larsson-lab/SEISMIC              #
+#           v0.1.0 commit: 688e961            #
+#                                             #
+#                                             #
+#         Modified for IntOGen use on:        #
+#                Jan 30,2024                  #
+#                                             #
+#      Applied patch available on GitHub:     #
+#       bbglab/containers/intogen/seismic     #
+#                                             #
+###############################################
+
 suppressPackageStartupMessages(library(fitdistrplus))
 suppressPackageStartupMessages(library(tidyverse))
 suppressPackageStartupMessages(library(plyranges))
@@ -15,25 +29,23 @@ suppressPackageStartupMessages(library(cowplot))
 
 main <- function(){
   args <- commandArgs(trailingOnly = TRUE)
-  config <- read_yaml(args[1])
-  # Override the number of cores in config if desired. This does not change the results, but one might want to change it depending on server usage, without fiddling with config files.
-  if(length(args) > 1){
-    no_of_cores <- as.integer(args[2])
-  } else {
-    no_of_cores <- config$cores
-  }
+  config <- read_yaml(args[1]) # First arg is the path to the config file
   
-  if(length(args) > 2){
-    config$cancer_type <- args[3]
-  }
+  config$mutations_path <- args[2]
+  config$out_base_name <- str_split_i(config$mutations_path, "\\.", 1)
+  config$cancer_type <- args[3]
+  config$seq_type <- args[4]
   
-  print_logo()
+  config$reference_genome <- args[5]
+  no_of_cores <- as.integer(args[6])
+     
+  #print_logo()
   
   # If we are not dealing with CDSs, all mutation effects will be 'na', so change effects_to_keep to that to not filter anything out.
   if(!config$annotate_cds_effects) config$effects_to_keep <- 'na'
   
   genome <- load_bsgenome(config$reference_genome)
-  fig_dir <- paste0(config$out_dir, '/figures/')
+  fig_dir <- paste0(config$out_dir, 'figures/')
   system(paste("mkdir -p", config$out_dir, fig_dir))
   
   cat("Loading regions\n")
@@ -166,7 +178,22 @@ main <- function(){
   mut_effects.df <- split.data.frame(mut_effects.df, mut_effects.df$test_region)
   
   if(length(test_region_list) == 0){
-    cat("No regions left for testing after filtering. Quitting\n")
+    cat("No regions left for testing after filtering.\n")
+    cat("Writing empty file...\n")
+    
+    empty.df <- data.frame(
+        region = character(0),
+        mutated_tumours = integer(0),
+        exp_mutated_tumours = integer(0),
+        p.cohort_skew = numeric(0),
+        fdr.cohort_skew = numeric(0)
+    )
+
+    empty.df %>%
+      write_tsv(paste0(config$out_dir,
+                       config$out_base_name, ".", "seismic.tsv"))
+    
+    cat("Quitting\n")
     quit()
   }
   
@@ -239,7 +266,7 @@ main <- function(){
       }
       
       mutprobs_per_combined_muttype.df <- calc_mutprobs_per_combined_muttype(test_region_mut_effects_tidy.df, combined_mutfreqs.df, config$effects_to_keep)
-      
+
       
       # Filter mutations to only look at those overlapping the test regions (e.g., the CDSs in a gene)
       mutations.df <- effect_filtered_mutations.gr %>% 
@@ -1782,28 +1809,28 @@ get_cds_seq <- function(test_regions.gr, test_region_arg, genome){
     paste0(collapse = '')
 }
 
-print_logo <- function(){
-logo <- r"{  
-                                                  _--   --_          
-                                               _/           \_       
-                                             /       ___       \     
-                                            /      -     -      \    
-                     ***                   /     /   ***   \     \   
-                    *****                  |    |   *****   |    |   
-                     ***                   \     \   ***   /     /   
-                                            \      -     -      /    
-  ****     *******   ***     ****     **        **   ***      ****   
-***  ***   *******   ***   ***  ***   ***      ***-__***__--******** 
-***   **   **        ***   ***   **   ****    ****   ***   ***    ***
- ****      *****     ***    ****      *****  *****   ***   ***       
-   ****    *****     ***      ****    *** **** ***   ***   ***       
-**   ***   **        ***   **   ***   ***  **  ***   ***   ***    ***
-***  ***   *******   ***   ***  ***   ***      ***   ***    ******** 
-  ****     *******   ***     ****     ***      ***   ***      ****   
+# print_logo <- function(){
+# logo <- r"{  
+#                                                   _--   --_          
+#                                                _/           \_       
+#                                              /       ___       \     
+#                                             /      -     -      \    
+#                      ***                   /     /   ***   \     \   
+#                     *****                  |    |   *****   |    |   
+#                      ***                   \     \   ***   /     /   
+#                                             \      -     -      /    
+#   ****     *******   ***     ****     **        **   ***      ****   
+# ***  ***   *******   ***   ***  ***   ***      ***-__***__--******** 
+# ***   **   **        ***   ***   **   ****    ****   ***   ***    ***
+#  ****      *****     ***    ****      *****  *****   ***   ***       
+#    ****    *****     ***      ****    *** **** ***   ***   ***       
+# **   ***   **        ***   **   ***   ***  **  ***   ***   ***    ***
+# ***  ***   *******   ***   ***  ***   ***      ***   ***    ******** 
+#   ****     *******   ***     ****     ***      ***   ***      ****   
                                                                      
-}"
-cat(logo)
-}
+# }"
+# cat(logo)
+# }
 
 ################################################
 # Run the main function
diff --git a/config/config.yaml b/config/config.yaml
new file mode 100644
index 0000000..83785da
--- /dev/null
+++ b/config/config.yaml
@@ -0,0 +1,46 @@
+mutations_path: "input"
+test_regions_path: "/seismic/data/cds_chr1-22_XY.tsv" #? or cds.regions.tsv? 
+
+out_dir: "" 
+out_base_name: "cohort"
+
+reference_genome: "refgenome" #h19 or h38
+seq_type: "WXS" 
+sig_type: "cohort"
+cancer_type: "TUMOR"
+
+annotate_cds_effects: TRUE
+effects_to_keep:
+  - "m"
+  - "n"
+
+# WXS-specific
+scale_exp_to_repl_reg: FALSE
+scale_exp_to_expr_reg: FALSE
+
+wxs_covered_regions_path: "/seismic/data/cds_chr1-22_XY.tsv"
+
+# Which tests to perform
+# Turn scaling back on if we run the recurrence tests
+test_recurrence_alone: FALSE
+test_cohort_skew_alone: TRUE
+test_combined_model: FALSE
+
+# WGS-specific
+scale_exp_with_silent_muts: FALSE
+# silent_mut_flank_upstream: 10000
+# silent_mut_flank_downstream: 10000
+# min_silent_mutations: 10
+
+
+plotting_max_p: 0.001
+# plot_region_list:
+#   - "AKT1"
+plot_region_list_only: FALSE 
+
+# Filtering test regions
+remove_overlapping_test_regions: FALSE
+min_mutations: 3
+
+no_of_simulations: 10000
+cores: 8
diff --git a/demo/install_R_dependencies.R b/demo/install_R_dependencies.R
index dc1e81b..b6be510 100644
--- a/demo/install_R_dependencies.R
+++ b/demo/install_R_dependencies.R
@@ -1,6 +1,22 @@
 cat('Installing R dependencies\n')
-install.packages(c('tidyverse', 'yaml', 'foreach', 'doSNOW', 'stringi', 'inline', 'Rcpp', 'fitdistrplus', 'data.table', 'BiocManager', 'cowplot', 'R.utils'))
+options(repos = c(REPO_NAME = "https://packagemanager.rstudio.com/cran/latest"), timeout=max(1000, getOption("timeout")))
+
+install.packages('BiocManager', version='3.16', dependencies=TRUE)
+
 BiocManager::install("plyranges")
 BiocManager::install("BSgenome.Hsapiens.UCSC.hg19")
 BiocManager::install("BSgenome.Hsapiens.UCSC.hg38")
 
+install.packages('tidyverse', version='1.3.1', dependencies=TRUE)
+install.packages('yaml', version='2.3.5')
+install.packages('foreach', version='1.5.2')
+install.packages('doSNOW', version='1.0.20')
+install.packages('stringi', version='1.7.6')
+install.packages('inline', version='0.3.19')
+install.packages('Rcpp', version='1.0.8.3')
+install.packages('fitdistrplus', version='1.1.8')
+install.packages('data.table', version='1.14.2')
+install.packages('cowplot', version='1.1.1')
+install.packages('R.utils', version='2.12.2')
+install.packages(c('textshaping','ragg'), dependencies=TRUE)
+
